src/UnitTests/NullBehavior.cs

```diff
@@ -82,6 +84,12 @@ public void Should_return_null_for_nullable_properties()
             }
 
             [Fact]
+            public void Should_return_not_null_for_nullable_properties_that_are_user_defined()
+            {
+                _result.NullableMapFrom.ShouldEqual(0);

```

Comment:
So in this case I think the test is wrong, the result should be null. In
the original issue, it should be what was expected there too.

On Thursday, July 7, 2016, Tyler Carlson <notifications@github.com> wrote:

> In src/UnitTests/NullBehavior.cs
> <https://github.com/AutoMapper/AutoMapper/pull/1485#discussion_r70010586>:
>
> > @@ -82,6 +84,12 @@ public void Should_return_null_for_nullable_properties()
> >              }
> >
> >              [Fact]
> > +            public void Should_return_not_null_for_nullable_properties_that_are_user_defined()
> > +            {
> > +                _result.NullableMapFrom.ShouldEqual(0);
>
> I refer to line 49. Cause it was user defined from MapFrom, ala a
> CustomExpression we use the MapFrom's expression's return type which is int.
>
> Yea it's not the best example. The issue is if I put conditions ? : into
> line 49 the seal would throw an exception. So I'm just saying if user
> defined it it's possible they are there and therefore we can't assume the
> nullable return type will work. So tests just say it doesn't null on user
> defined. Again not the best of tests.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/AutoMapper/AutoMapper/pull/1485/files/fe6ad9f72e5708aa17d73d7a48191888e4432a6f#r70010586>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAGYMn0j2_Gi9ffwg-V4HYBSwGICAQhXks5qTZnAgaJpZM4JHf0X>
> .
>